frontend/public/lib/imageOperations.ts

```diff
@@ -131,13 +131,32 @@ const drawImageOnCanvas = (image, canvas) => {
   }
 };
 
-export async function blobFromObjectUrl(objectUrl) {
-  var a = new FileReader();
-  const blob = await fetch(objectUrl).then((r) => r.blob());
-  return new Promise(function (resolve) {
-    a.onload = function (e) {
-      resolve(e.target!.result);
+export async function blobFromObjectUrl(objectUrl: string): Promise<string> {
+  try {
+    const response = await fetch(objectUrl);
+    const originalBlob = await response.blob();
+
+    const options = {
+      maxSizeMB: 0.5,
+      useWebWorker: true,
     };
-    a.readAsDataURL(blob);
-  });
+
+    // imageCompression accepts Blob despite its TypeScript definition requiring File
+    const compressedBlob = await imageCompression(originalBlob as File, options);
```

Comment:
File is based on Blob, so they are partly compatible. Although looking at the imageCompression implementation it does use some file specific properties like modified date. Obviously without negative side effects.

https://github.com/Donaldcwl/browser-image-compression has not been maintained since 2023. We need to add a task to find an alternative.